typescript/sdk/src/contracts/contracts.ts

```diff
@@ -146,6 +146,15 @@ export function attachContractsMapAndGetForeignDeployments<
   };
 }
 
+export function attachAndConnectContracts<F extends HyperlaneFactories>(
+  addresses: HyperlaneAddresses<F>,
+  factories: F,
+  connection: Connection,
+): HyperlaneContracts<F> {
+  const contracts = attachContracts(addresses, factories);
+  return connectContracts(contracts, connection);
+}
```

Comment:
saw your comment on a commit and moving into the PR @yorhodes 
> also how much does this function really save us in verbosity?

I like having it to reduce having an extra "const x = BLAH" before assigning to factories
<img width="549" alt="image" src="https://github.com/hyperlane-xyz/hyperlane-monorepo/assets/10051819/14108dfc-96df-45c7-9722-7e042c8eb093">